rq/queue.py

```diff
@@ -289,15 +289,20 @@ def enqueue_dependents(self, job):
         # TODO: can probably be pipelined
         from .registry import DeferredJobRegistry
 
-        registry = DeferredJobRegistry(self.name, self.connection)
+
 
         while True:
             job_id = as_text(self.connection.spop(job.dependents_key))
             if job_id is None:
                 break
             dependent = self.job_class.fetch(job_id, connection=self.connection)
+            registry = DeferredJobRegistry(dependent.origin, self.connection)
             registry.remove(dependent)
```

Comment:
Can you also pipeline the `registry.remove` and `queue.enqueue_job` methods? In addition to being more performant, this also protects us from having the job already removed `DeferredJobRegistry`, but not properly enqueued in case something fails (usually network).